tests/test_pyglmnet.py

```diff
@@ -388,3 +385,47 @@ def test_random_state_consistency():
 
     assert_array_equal(ypred_a, ypred_b)
     assert_array_equal(ypred_b, ypred_c)
+
+
+@pytest.mark.parametrize("distr", ALLOWED_DISTRS)
+def test_simulate_glm(distr):
+    """Test that every generative model can be simulated from."""
+
+    n_samples, n_features = 10, 3
+
+    # sample random coefficients
+    beta0 = np.random.normal(0.0, 1.0, 1)
+    beta = np.random.normal(0.0, 1.0, n_features)
+
+    np.random.seed(1)
+    X = np.random.normal(0.0, 1.0, [n_samples, n_features])
+    simulate_glm(distr, beta0, beta, X)
+
+    # If the distribution name is garbage it will fail
+    distr = 'multivariate_gaussian_poisson'
+    with pytest.raises(ValueError):
+        simulate_glm(distr, 1.0, 1.0, np.array([[1.0]]))
+
+
+@pytest.mark.parametrize("y_func",
+                         [list, tuple, np.array,
+                          lambda x: np.array(x).reshape(-1, 1)])
```

Comment:
see comment above. Probably better to remove this case of y as a 2D array. I think it could lead to weird corner cases that we should avoid

Reply:
Changed this test, only allowing ndarrays for both X and y now.